docs/acme-divergences.md

```diff
@@ -0,0 +1,82 @@
+# Boulder divergences from ACME
+
+While Boulder attempts to implement the ACME specification as strictly as possible
+there are places at which we will diverge from the letter of the specification for
+various reason.
```

Comment:
One higher level question I have is whether we want to document the "SHOULD" divergences the same way we document the others. 

E.g. Not sending a `Retry-After` header for rate limiting is within our "rights" as implementors of ACME based on the RFC2119 reading of "should". This seems like a different category of choice than something like using a different error namespace.